automation_tools/satellite6/log.py

```diff
@@ -0,0 +1,105 @@
+    Running some process, could be Satellite Upgrade
+    [root@sathost.redhat.com] Executing task 'get_line_count'
+    [root@sathost.redhat.com] Executing task 'fetch_appended_log_lines'
+    ### No changes in /var/log/foreman-proxy/proxy.log
+    ### No changes in /var/log/foreman/production.log
+    ### No changes in /var/log/foreman-installer/satellite.log
+    ### No changes in /var/log/candlepin/candlepin.log
+    ### No changes in /var/log/tomcat/catalina.out
+    ### No changes in /var/log/messages
+    ### No changes in /var/log/mongodb/mongodb.log
+
+    """
+
+    def __init__(self, sat_host):
+        """Initializes context manager with Satellite hostname
+
+        :param sat_host: str
+        """
+        self.sat_host = sat_host
+        self.log_state = {
+            '/var/log/foreman-installer/satellite.log': 0,
+            '/var/log/foreman/production.log': 0,
+            '/var/log/foreman-proxy/proxy.log': 0,
+            '/var/log/candlepin/candlepin.log': 0,
+            '/var/log/messages': 0,
+            '/var/log/mongodb/mongodb.log': 0,
+            '/var/log/tomcat/catalina.out': 0
+        }
+
+    def __enter__(self):
+        """
+        Fetch current line count for Satellite log files
+        :return: LogAnalyzer
+        """
+        self._update_log_files_state()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Analyzes log files checking if some error occurred since last
+        log_state_dct
+        """
+        self._update_log_files_state()
+
+        def analyze(log_file, content):
+            """Analyzes appended content from log file. For now it is only
+            printing content
+            """
+            print('### Analyzing %s:' % log_file)
+            print(content)
+
+        def fetch_appended_log_lines():
+            for log_file, lines_appended in self.log_state.items():
+                if lines_appended > 0:
+                    content = run(
+                        'tail -n {} {}'.format(lines_appended, log_file),
+                        quiet=True
+                    )
+                    analyze(log_file, content)
+                else:
+                    print('### No changes in %s' % log_file)
+
+        execute(fetch_appended_log_lines, host=self.sat_host)
+
+    def _update_log_files_state(self):
+        """Update log_dct with adding delta from current number of lines of
+        each
+        item and the last provided by dct. Ex:
+
+        So this method can be used to check how many lines were appended on
+        a file
+        during some processes and used to tail them. If log dct is None a
+        new dict
+        is created
+
+        :param sat_host: str with satellite hostname
+        :param log_dct: dict to be updated
+        """
+
+        def get_line_count():
+            for log_file, old_value in self.log_state.items():
+                current_value = int(
```

Comment:
As per the code, I understood:
When upgrade will start with the context manager, initially, the log_state[log_file] = current_value(an example that in some logfile there will 500 line already then this current value will become 500) - old_value(which will be 0) = 500

Then, that 500 number will be sent to fetch_appended_log_files() def and the content = tail -n 500 <logfilename>, which will have last 500 line in that file, which will be sent to analyze() def to print which wiil really be mess an unecessary logs regardless to upgrade/current process.

I may understood it wrong, please correct me i m wrong.